src/System.Private.Uri/src/System/PooledCharArray.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+
+namespace System
+{
+    internal struct PooledCharArray
```

Comment:
We have several types like this already in corefx.  It would be nice not to add yet another.  This looks an awful lot like ValueStringBuilder.  Please try to use that instead.

Reply:
(I will refrain from commenting on this implementation of this type, as I'm hopeful it'll go away.  If for some reason it can't, there are a bunch of things that should be looked at, e.g. why is it manually copying in a bunch of places rather than using Array.Copy, why is it copying the original destination array, etc.)